lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb

```diff
@@ -0,0 +1,139 @@
+require 'fileutils'
+
+Puppet::Type.type(:bootstrap_rpm).provide(:bootstrap_rpm) do
+
+  commands :rpmbuild => 'rpmbuild'
+
+  def create
+    base_dir
+    copy_sources
+    write_specfile
+    build_rpm
+    copy_rpm
+    link_rpm
+  ensure
+    FileUtils.remove_dir(base_dir) if File.directory?(base_dir)
```

Comment:
To get around the `SOURCES` issue I had to use:
```suggestion
    FileUtils.remove_dir(base_dir) if File.directory?(base_dir)
    @base_dir = nil
```

This does suggest `create` is called more than once. Likely that `exists?` is incorrect.

Reply:
Yes, its the refresh triggering it as the full idempotency of this is not accounted for yet. I wanted to get draft feedback on the overall design approach before solving the complex issue of detecting when a new RPM is needed.